src/ui/blockUi-editor.js

```diff
@@ -114,6 +118,145 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
+            }
+        },
+        selectors: {
+            mediaPreview: ".sjrkc-st-block-media-preview",
+            progressArea: ".sjrkc-st-file-share-progress",
+            responseArea: ".sjrkc-st-file-share-response",
+            responseText: ".sjrkc-st-file-share-response-text",
+            singleFileUploader: ".sjrkc-st-block-uploader-input",
+            uploadButton: ".sjrkc-st-block-media-upload-button"
+        },
+        invokers: {
+            setServerResponse: {
+                this: "{that}.dom.responseText",
+                method: "text",
+                args: ["{arguments}.0"]
+            },
+            updateMediaPreview: null // to be specified by implementing grades
+        },
+        events: {
+            onMediaUploadRequested: null
+        },
+        listeners: {
+            "{templateManager}.events.onTemplateRendered": {
+                this: "{that}.dom.uploadButton",
+                method: "click",
+                args: ["{that}.events.onMediaUploadRequested.fire"],
+                namespace: "bindOnMediaUploadRequested"
+            }
+        },
+        components: {
+            // handles uploading a single file for storage
+            singleFileUploader: {
+                type: "sjrk.storyTelling.block.singleFileUploader",
+                createOnEvent: "{templateManager}.events.onTemplateRendered",
+                container: "{editor}.dom.singleFileUploader",
+                options: {
+                    modelRelay: {
+                        "uploadState": {
+                            target: "{editor}.model",
+                            singleTransform: {
+                                type: "fluid.transforms.valueMapper",
+                                defaultInputPath: "uploadState",
+                                match: {
+                                    "ready": {
+                                        outputValue: {
+                                            previewVisible: true,
+                                            progressAreaVisible: false,
+                                            responseAreaVisible: false,
+                                            uploadButtonDisabled: false
+                                        }
+                                    },
+                                    "uploading": {
+                                        outputValue: {
+                                            previewVisible: false,
+                                            progressAreaVisible: true,
+                                            responseAreaVisible: false,
+                                            uploadButtonDisabled: true
+                                        }
+                                    },
+                                    "errorReceived": {
+                                        outputValue: {
+                                            previewVisible: true,
```

Comment:
Shall the preview be visible when the upload receives an error?

Reply:
The designs for [SJRK-396](https://issues.fluidproject.org/browse/SJRK-396) seem to suggest that the preview will be hidden when the upload receives an error. I will set the value to `false` for now, and we can review the question when SJRK-396 is picked up.